packages/core/src/components/popover/popover.scss

```diff
@@ -0,0 +1,94 @@
+@use '../../helpers';
+@use '../../mixins';
+@use 'sass:map';
+
+@mixin Popover() {
+  @if not mixins.includes('Popover') {
+    @include _Popover();
+  }
+}
+
+@mixin _Popover() {
+  .ods-popover-anchor {
+    pointer-events: none;
+    position: absolute;
+    z-index: 9999999;
+  }
+
+  .ods-popover {
+    height: fit-content;
+    max-width: helpers.space(50);
+    position: absolute;
+    width: max-content;
+
+    // position ==========>
+    &.-top-center,
+    &.-top-start,
+    &.-top-end {
+      @include _positionFrom(bottom);
+    }
+
+    &.-bottom-center,
+    &.-bottom-start,
+    &.-bottom-end {
+      @include _positionFrom(top);
+    }
+
+    &.-left-center,
+    &.-left-start,
+    &.-left-end {
+      @include _positionFrom(right);
+    }
+
+    &.-right-center,
+    &.-right-start,
+    &.-right-end {
+      @include _positionFrom(left);
+    }
+    // <========== position
+
+    // alignment ==========>
+    &.-top-center,
+    &.-bottom-center {
+      @include _center(X);
+    }
+
+    &.-left-center,
+    &.-right-center {
+      @include _center(Y);
+    }
+
+    &.-top-start,
+    &.-bottom-start {
+      left: 0;
+    }
+
+    &.-left-start,
+    &.-right-start {
+      top: 0;
+    }
+
+    &.-top-end,
+    &.-bottom-end {
+      right: 0;
+    }
+
+    &.-left-end,
+    &.-right-end {
+      bottom: 0;
+    }
+    // <========== alignment
+  }
+}
+
+@mixin _positionFrom($side) {
```

Comment:
I like this solution in the end.

Reply:
Thanks for helping me iterate on it, the Sass loop looked horrendous, as did the verbose one-by-one definitions 😂